cmd/subctl/diagnose.go

```diff
@@ -255,6 +269,10 @@ func diagnoseAll(clusterInfo *cluster.Info, status reporter.Interface) bool {
 
 	fmt.Println()
 
+	success = diagnose.ServiceDiscovery(clusterInfo, status) && success
```

Comment:
We should check if service discovery is installed. Also I think we should move this after all the connectivity components to L291 below.

```suggestion
        if clusterInfo.ServiceDiscovery != nil {
            success = diagnose.ServiceDiscovery(clusterInfo, status) && success
         } else {
	    status.Success("Service discovery is not installed - skipping")
	}
```